### PR TITLE
feat: add Windows Consumption to Flex Consumption migration workflow

### PR DESCRIPTION
**Pull request by mathewc**
## Summary

Adds a guided Azure Functions migration workflow for moving apps from **Windows Consumption (Y1/Dynamic)** to **Flex Consumption (FC1/Linux)**. This is peer to the existing Linux Consumption upgrade instructions.

The Windows to Flex manual migration guide [here](https://learn.microsoft.com/en-us/azure/azure-functions/migration/migrate-plan-consumption-to-flex?tabs=azure-cli%2Ccopilot-cli%2Csystem-assigned%2Ccontinuous%2Cbicep%2Ctraces-table&pivots=platform-windows) was the starting point. This skill automates the CLI instructions of that guide. It also goes beyond that to improve and automate the experience.

Unlike the existing Linux migration path, this workflow accounts for the Windows-to-Linux platform change, including Windows-specific code dependencies, runtime compatibility, deployment model differences, trigger safety, configuration conversion, and controlled production cutover.

The source app remains read-only during assessment and replication. Any source-changing cutover, rollback, or cleanup action requires explicit, action-specific user approval.

## Sample prompts

Users can initiate the workflow with prompts such as:

- `Upgrade my Windows Function App <app-name> from Consumption to Flex Consumption.`
- `Assess whether <app-name> is ready to migrate from Windows Consumption to Flex Consumption.`
- `Migrate <app-name> to Flex Consumption as <new-app-name>.`
- `Migrate <app-name> to Flex Consumption with a new storage account.`
- `Migrate <app-name> to Flex Consumption using storage account <storage-name>.`
- `What Windows Consumption Function Apps can I upgrade to Flex Consumption?`
- `Resume the Flex Consumption migration for <app-name>.`

## Upgrade workflow

The skill performs the migration as a resumable, side-by-side workflow:

1. **Discover and assess**
   - Identify the source app, resource group, region, runtime, and plan.
   - Confirm Flex Consumption availability and runtime support.
   - Detect unsupported deployment slots, certificates, polling Blob triggers, and other platform limitations.
   - Locate source code or retrieve the deployed package when source is unavailable.
   - Scan for Windows-specific dependencies such as Win32 APIs, COM, WMI, registry access, Windows certificate-store access, Windows-only packages, and Windows RID builds.
   - Generate a structured upgrade assessment report.

2. **Inventory source configuration**
   - Catalog app-setting names without exposing values.
   - Inventory site configuration, identities, RBAC assignments, CORS, domains, access restrictions, authentication, certificates, triggers, and deployment configuration.
   - Catalog explicit Azure Files mount metadata as not migrated because Windows Consumption doesn't support these mounts.

3. **Prepare application code**
   - Identify required Windows-to-Linux compatibility changes.
   - Detect unsupported runtimes and deployment formats.
   - Require polling Blob triggers to use Event Grid.
   - Build or otherwise validate the prepared application before deployment.

4. **Select target resources**
   - Propose a globally unique target app name, defaulting to `<source>-flex`.
   - Select whether to reuse source storage, create a new storage account, or use a specified existing account.
   - Optionally configure identity-based runtime storage.

5. **Create and configure the Flex app**
   - Create the FC1 Linux Function App alongside the source.
   - Safely copy applicable app settings while preserving target-owned settings.
   - Convert hierarchical setting names from `:` to `__` for Linux and detect ambiguous mappings.
   - Migrate compatible site configuration, managed identities, RBAC assignments, CORS, custom domains, and access restrictions.
   - Preserve and verify the target Application Insights configuration.
   - Identify configuration requiring manual handling, such as built-in authentication and site-scoped certificates.

6. **Establish trigger safety**
   - Classify every function trigger before deployment.
   - Determine whether target dependencies are isolated from production.
   - Offer controlled handling for non-HTTP functions that could immediately consume events or run schedules.
   - Account for Linux's inability to disable function names containing hyphens through per-function app settings.
   - Record user decisions and safety state in the migration status artifact.

7. **Deploy code**
   - Support updating CI/CD, publishing a local project with Functions Core Tools, or deploying the retrieved source package.
   - Block deployment when required target settings or trigger-safety decisions are incomplete.
   - Verify function synchronization and effective disablement after deployment.

8. **Validate the target**
   - Smoke-test HTTP endpoints when network access permits.
   - Validate runtime, configuration, identities, monitoring, and non-HTTP trigger behavior.
   - Record tests as deferred rather than weakening intentional access restrictions.
   - Keep the original Windows app available for rollback.

9. **Cut over safely**
   - Enable and validate non-HTTP functions individually.
   - Require immediate approval before disabling a corresponding source function.
   - Support source-first transactions where Linux function-name limitations prevent target-side disablement.
   - Validate concurrent-processing risks and rollback behavior.

10. **Clean up or roll back**
    - Offer source cleanup only after production traffic and triggers have been cut over and validated.
    - Require explicit approval before stopping, modifying, or deleting the source app.
    - Preserve resumable progress and rollback instructions throughout the migration.

## Additional changes

- Separates the existing Linux Consumption migration references from the new Windows-specific workflow.
- Adds persistent migration progress and artifact-handling guidance.
- Adds idempotent and resumable automation instructions.
- Preserves target-created storage and Application Insights settings during app-setting migration.
- Documents package retrieval from Windows Consumption content storage.
- Removes target Azure Files mount recreation; mount metadata is inventory-only.

## End-to-end test migration

This workflow was developed and exercised against `mathewc-winflex-kitchensink`, a purpose-built Windows Consumption Function App containing representative Azure features and migration edge cases.

The test app includes:

- .NET 8 isolated worker on Windows Consumption.
- HTTP, Timer, Storage Queue, Blob, Service Bus, Event Hubs, and Cosmos DB triggers.
- A polling Blob trigger requiring conversion to Event Grid.
- System-assigned and user-assigned managed identities.
- RBAC assignments for Storage, Event Hubs, and Application Insights.
- Connection-string and identity-based bindings.
- AAD-authenticated Application Insights ingestion.
- CORS, HTTP/2, HTTPS-only, TLS, access restrictions, and a custom scale limit.
- Source deployment content stored on the Windows Azure Files content share.
- Explicit Azure Files mount metadata for testing inventory behavior.
- Hierarchical app settings containing `:`, including `FunctionsHostingConfig:CustomSetting`.
- Intentional Windows compatibility fixtures covering Windows-only APIs/packages, Windows RID publishing, hard-coded Windows paths, and Windows command execution.

The test migrated the app side-by-side to `mathewc-winflex-kitchensink-flex` on FC1 in the same region.

### Scenarios verified

- Generated and reviewed the structured compatibility assessment.
- Located and scanned the local source project.
- Detected the intentional Windows compatibility fixtures.
- Applied the source compatibility fixes and rebuilt successfully.
- Converted the polling Blob trigger to Event Grid mode.
- Explicitly selected the target storage strategy.
- Created the Linux Flex Consumption target.
- Copied filtered settings without exposing values.
- Converted `FunctionsHostingConfig:CustomSetting` to `FunctionsHostingConfig__CustomSetting`.
- Repeated the settings transfer to verify idempotency.
- Preserved target-owned storage and Application Insights settings.
- Recreated managed identities, RBAC assignments, CORS, access restrictions, monitoring configuration, and the scale limit.
- Classified and disabled six non-HTTP functions before deployment to prevent unintended processing.
- Published the application using Functions Core Tools.
- Verified all seven functions synchronized successfully.
- Verified the HTTP function returned `200` and reported Linux/.NET 8.
- Exercised the fallback package-download path from the Windows content share.
- Downloaded and validated a 17 MB deployment ZIP containing 146 entries.
- Confirmed explicit Azure Files mount metadata is cataloged but not migrated.
- Left the original Windows app running for rollback.

The test concluded as a successful validation-only migration. Production-connected non-HTTP triggers were intentionally left disabled because a production cutover was outside the test scope.

### Repository validation

- Verified workflow numbering is contiguous from `4a` through `4k`, with optional `4b.opt`.
- Verified no stale or unresolved step references remain.
- Ran the markdown reference validator successfully for all skills.
- Ran `git diff --check` successfully.

### CODE REVIEW

File: plugins/azure-skills/skills/azure-upgrade/references/services/functions/consumption-to-flex/linux/workflow.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 > **Source**: Azure Functions Consumption Plan (Y1/Dynamic) on Linux
```
Comment (mathewc) [inline plugins/azure-skills/skills/azure-upgrade/references/services/functions/consumption-to-flex/linux/workflow.md:3]: Because I moved/renamed this file, we'll need to update the [[Azure Skills in GitHub Copilot](https://github.com/microsoft/GitHub-Copilot-for-Azure/blob/main/plugins/azure-skills/skills/azure-upgrade/references/services/functions/consumption-to-flex.md) link in the "Migration Methods" table of [Migrate Consumption plan apps to the Flex Consumption plan](https://learn.microsoft.com/en-us/azure/azure-functions/migration/migrate-plan-consumption-to-flex?tabs=github-copilot%2Ccopilot-cli%2Csystem-assigned%2Ccontinuous%2Cbicep%2Ctraces-table&pivots=platform-linux) documentation. We'll also want to update that table to show a green check for Windows support since we're adding it here.

File: plugins/azure-skills/skills/azure-upgrade/references/services/functions/consumption-to-flex/windows/automation/discovery-and-assessment.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+### 2e. Check TLS/SSL Certificates
+
+List certificates available in the source app's resource group:
+
+```bash
+az webapp config ssl list --resource-group <RESOURCE_GROUP> --output table
+```
+
+If this returns output, determine which certificates the source app actually uses. Record whether each used certificate is private or public and confirm the target can fit within the Flex Consumption limit of three private and three public certificates per app. Plan to re-add them using the documented [site-scoped certificate process](https://learn.microsoft.com/en-us/azure/azure-functions/flex-consumption-how-to#configure-site-scoped-certificates).
```
Comment (mathewc) [inline plugins/azure-skills/skills/azure-upgrade/references/services/functions/consumption-to-flex/windows/automation/discovery-and-assessment.md:99]: Currently this skill doesn't do an auto migration of these certificates. It catalogs them and instructs the user, pointing to our published instructions. TBD on whether we want to go further.
